### PR TITLE
Add `ready` field for ticdc's status (#4273)

### LINKED ISSUE #4263: Update the status of TiCDC
## Feature Request

**Is your feature request related to a problem? Please describe:**

Currently, the following fields are included in the TiCDC status:
```
// TiCDCStatus is TiCDC status
type TiCDCStatus struct {
	Synced      bool                    `json:"synced,omitempty"`
	Phase       MemberPhase             `json:"phase,omitempty"`
	StatefulSet *apps.StatefulSetStatus `json:"statefulSet,omitempty"`
	Captures    map[string]TiCDCCapture `json:"captures,omitempty"`
}
```
And TiDB Operator checks the Pod status [here](https://github.com/pingcap/tidb-operator/blob/master/pkg/manager/member/ticdc_member_manager.go#L225) without considering if the Pod exists or not.

So we should:
1. Check if the Pod exists first and only add capture info if the Pod exist so that users are able to know if the Pod has been created or not
2. Add a new field `state` in the `type TiCDCCapture struct` to indicate whether the capture is ready or not.
 
**Describe the feature you'd like:**

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #4273 to release-1.2
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In tidb-operator repo:
git pr https://github.com/pingcap/tidb-operator/pull/4292
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/tidb-operator.git pr/4292:release-1.2-b339f41b7b04
```

---

### What problem does this PR solve?

Close #4263 

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [x] Manual test 

Apply Cluster and check status\

- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

File: deploy/gcp/manifests/crd.yaml [hunk 1]
```diff
@@ -1 +1 @@
-../../../manifests/crd.yaml
```
Comment (DanielZhangQD) [inline deploy/gcp/manifests/crd.yaml:1]: What's the change here?

Comment (DanielZhangQD) [reply deploy/gcp/manifests/crd.yaml:1]: The following files should be removed:
manifests/crd/v1/pingcap.com_tidbclusters.yaml
manifests/crd/v1beta1/pingcap.com_tidbclusters.yaml

Comment (KanShiori) [reply deploy/gcp/manifests/crd.yaml]: done

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Pull request merged**